--- a/lib/regcomp.c	2020-11-24 17:06:08.159223858 +0000
+++ b/lib/regcomp.c	2020-11-24 17:06:15.630253923 +0000
@@ -3808,11 +3808,7 @@
 create_tree (re_dfa_t *dfa, bin_tree_t *left, bin_tree_t *right,
 	     re_token_type_t type)
 {
-  re_token_t t;
-#if defined GCC_LINT || defined lint
-  memset (&t, 0, sizeof t);
-#endif
-  t.type = type;
+  re_token_t t = { .type = type };
   return create_token_tree (dfa, left, right, &t);
 }
 
